### PR TITLE
Add clarifications to the docs regarding `to_feather`

### PR DESCRIPTION
**Pull request by jmakov**
- [X] closes https://github.com/pandas-dev/pandas/issues/28208
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [ ] whatsnew entry

As discussed in https://github.com/pandas-dev/pandas/issues/28208, the docs should be more precise about in which cases `to_feather()` will fail.

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -2552,7 +2552,8 @@ def to_stata(
     @deprecate_kwarg(old_arg_name="fname", new_arg_name="path")
     def to_feather(self, path: FilePathOrBuffer[AnyStr], **kwargs) -> None:
         """
-        Write a DataFrame to the binary Feather format.
+        Write a DataFrame to the binary Feather format. Requires a default index.
```
Comment (jreback) [inline pandas/core/frame.py:2555]: these changes should be in Notes. this should have failed the doc-string checker (as this is no longer a summary line). are we still running these checks, cc @MarcoGorelli

Comment (jmakov) [reply pandas/core/frame.py:2555]: @jreback can you please clarify what do you mean with Notes? I searched the repo for the docs and this is the only place I've found so I updated it. Do you perhaps mean that you want a summary line for the first sentence followed by an empty line with more details?

Comment (jreback) [reply pandas/core/frame.py:2555]: Notes section, you have to add it. yes summary is 1 line.

Comment (jmakov) [reply pandas/core/frame.py]: OK, added what I understand you mean by "notes section" - another paragraph. Let me know if there's something else I can do.

File: pandas/core/frame.py [hunk 2]
```diff
@@ -2554,6 +2554,11 @@ def to_feather(self, path: FilePathOrBuffer[AnyStr], **kwargs) -> None:
         """
         Write a DataFrame to the binary Feather format.
 
+        This function writes the dataframe as a `feather file
```
Comment (jreback) [inline pandas/core/frame.py:2557]: put *below* the Parameters section
its called
```
Notes
-------
<text>
```

Comment (jmakov) [reply pandas/core/frame.py]: Oh, didn't see that at all. Thanks.

**Comment by jmakov** [comment]
@jreback can this be merged?

**Pull request merged**